packages/jest-config/src/normalize.ts

```diff
@@ -556,10 +556,11 @@ export default async function normalize(
   }
 
   setupBabelJest(options);
-  // TODO: Type this properly
-  const newOptions = {
-    ...DEFAULT_CONFIG,
-  } as unknown as AllOptions;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const {moduleNameMapper, ...rest} = DEFAULT_CONFIG;
```

Comment:
@SimenB I think we can remove values like this instead of removing anything from default as users might using those values.